examples/gpio_generic.rs

```diff
@@ -0,0 +1,75 @@
+#![no_main]
+#![no_std]
+
+extern crate panic_rtt_target;
+
+use lpc8xx_hal::{
+    cortex_m_rt::entry,
+    delay::Delay,
+    gpio::{direction::Dynamic, GpioPin, Level},
+    pins::{DynamicPinDirection, GenericPin},
+    prelude::*,
+    CorePeripherals, Peripherals,
+};
+
+#[entry]
+fn main() -> ! {
+    rtt_target::rtt_init_print!();
+
+    // Get access to the device's peripherals. Since only one instance of this
+    // struct can exist, the call to `take` returns an `Option<Peripherals>`.
+    // If we tried to call the method a second time, it would return `None`, but
+    // we're only calling it the one time here, so we can safely `unwrap` the
+    // `Option` without causing a panic.
+    let cp = CorePeripherals::take().unwrap();
+    let p = Peripherals::take().unwrap();
+
+    // Initialize the APIs of the peripherals we need.
+    let mut delay = Delay::new(cp.SYST);
+
+    let gpio = {
+        let mut syscon = p.SYSCON.split();
+        p.GPIO.enable(&mut syscon.handle)
+    };
```

Comment:
Nitpick: The block is unnecessary and drops `syscon`, which anyone who wants to extend this example into something less trivial is going to need. This is probably a copy-past leftover from other GPIO examples who use this technique to paper over the differences between LPC82x and LPC845.

```suggestion
    let mut syscon = p.SYSCON.split();
    let gpio = p.GPIO.enable(&mut syscon.handle);
```

This is just a tiny complaint though. It can just as well stay as is.